commands/motor/MotorControl.java

```diff
@@ -14,25 +14,29 @@
  */
 public class MotorControl extends Command {
 	protected final Motor motor;
-	protected final Controller controller;
-	protected final int axis;
+	protected final Supplier<Double> speedSupplier;
 	protected final double scale;
+	protected final double offset;
 
 	/**
-	 * This Command directly controls a Motor based on an axis of the Controller.
+	 * This Command directly controls a Motor based on a double supplier;
 	 * This can allow an Operator to easily control a single Motor from an axis of the Controller.
 	 *
+	 * @param name
 	 * @param motor
 	 * @param controller
 	 * @param axis
 	 * @param scale
+	 * @param offset
+	 *        A constant to add to the motor's speed
+	 *        Useful for using a controller for fine-tuning a constant speed
 	 */
-	public MotorControl(Motor motor, Controller controller, int axis, double scale) {
-		super("MotorControl");
+	public MotorControl(String name, Motor motor, Supplier<Double> speedSupplier, double scale, double offset) {
+		super(name);
```

Comment:
This is pretty cool, however, we may run into compatibility problems with older versions. 10/10 would recommend for 2018-code, and maybe for 2017 if we take the time to do some refactoring.

Reply:
Compatibility is something we rarely consider as we are not supporting a huge number of people on historical versions. If older software relies on something, a branch can be created to support it, or the older software can adapt.

Reply:
Ok my B.